src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -519,30 +519,32 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
             [NotNull] PositionTagged<string> suffix,
             params AttributeValue[] values)
         {
+            bool useAttributeNameAsValue;
+
             if (values.Length == 0)
             {
                 // Explicitly empty attribute, so write the prefix
                 WritePositionTaggedLiteral(writer, prefix);
             }
-            else if (values.Length == 1 && values[0].Prefix == "" &&
-                (values[0].Value.Value is bool || values[0].Value.Value == null))
+            else if (!ShouldRenderAttribute(values, out useAttributeNameAsValue))
```

Comment:
This refactoring obscures what's going on. Suggest instead splitting into two methods that each return a `bool` e.g. `IsSingleBoolOrNullValue()` and `UseAttributeNameAsValue()`.